vsphere/resource_vsphere_host_virtual_switch_test.go

```diff
@@ -0,0 +1,253 @@
+				},
+				Providers:    testAccProviders,
+				CheckDestroy: testAccResourceVSphereHostVirtualSwitchExists(false),
+				Steps: []resource.TestStep{
+					{
+						Config: testAccResourceVSphereHostVirtualSwitchConfig(),
+						Check: resource.ComposeTestCheckFunc(
+							testAccResourceVSphereHostVirtualSwitchExists(true),
+						),
+					},
+				},
+			},
+		},
+		{
+			"basic, then remove a NIC",
+			resource.TestCase{
+				PreCheck: func() {
+					testAccPreCheck(tp)
+					testAccResourceVSphereHostVirtualSwitchPreCheck(tp)
+				},
+				Providers:    testAccProviders,
+				CheckDestroy: testAccResourceVSphereHostVirtualSwitchExists(false),
+				Steps: []resource.TestStep{
+					{
+						Config: testAccResourceVSphereHostVirtualSwitchConfig(),
+						Check: resource.ComposeTestCheckFunc(
+							testAccResourceVSphereHostVirtualSwitchExists(true),
+						),
+					},
+					{
+						Config: testAccResourceVSphereHostVirtualSwitchConfigSingleNIC(),
+						Check: resource.ComposeTestCheckFunc(
+							testAccResourceVSphereHostVirtualSwitchExists(true),
+						),
+					},
+				},
+			},
+		},
+		{
+			"standby with explicit failover order",
+			resource.TestCase{
+				PreCheck: func() {
+					testAccPreCheck(tp)
+					testAccResourceVSphereHostVirtualSwitchPreCheck(tp)
+				},
+				Providers:    testAccProviders,
+				CheckDestroy: testAccResourceVSphereHostVirtualSwitchExists(false),
+				Steps: []resource.TestStep{
+					{
+						Config: testAccResourceVSphereHostVirtualSwitchConfigStandbyLink(),
+						Check: resource.ComposeTestCheckFunc(
+							testAccResourceVSphereHostVirtualSwitchExists(true),
+						),
+					},
+				},
+			},
+		},
+		{
+			"basic, then change to standby with failover order",
+			resource.TestCase{
+				PreCheck: func() {
+					testAccPreCheck(tp)
+					testAccResourceVSphereHostVirtualSwitchPreCheck(tp)
+				},
+				Providers:    testAccProviders,
+				CheckDestroy: testAccResourceVSphereHostVirtualSwitchExists(false),
+				Steps: []resource.TestStep{
+					{
+						Config: testAccResourceVSphereHostVirtualSwitchConfig(),
+						Check: resource.ComposeTestCheckFunc(
+							testAccResourceVSphereHostVirtualSwitchExists(true),
+						),
+					},
+					{
+						Config: testAccResourceVSphereHostVirtualSwitchConfigStandbyLink(),
+						Check: resource.ComposeTestCheckFunc(
+							testAccResourceVSphereHostVirtualSwitchExists(true),
+						),
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testAccResourceVSphereHostVirtualSwitchCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tp = t
+			resource.Test(t, tc.testCase)
+		})
+	}
+}
+
+func testAccResourceVSphereHostVirtualSwitchPreCheck(t *testing.T) {
+	if os.Getenv("VSPHERE_HOST_NIC0") == "" {
+		t.Skip("set VSPHERE_HOST_NIC0 to run vsphere_host_virtual_switch acceptance tests")
+	}
+	if os.Getenv("VSPHERE_HOST_NIC1") == "" {
+		t.Skip("set VSPHERE_HOST_NIC1 to run vsphere_host_virtual_switch acceptance tests")
+	}
+	if os.Getenv("VSPHERE_ESXI_HOST") == "" {
```

Comment:
should this not be being caught in the test provider initialisation?

Reply:
This variable is not caught as it's not necessary on all tests, but it's necessary here.